policies/verify-instances-templates-reference/policies/rego_rule.rego

```diff
@@ -1,33 +1,32 @@
-
 package crownlabs_instance_template_reference
 
-t_ns := input.review.object.spec["template.crownlabs.polito.it/TemplateRef"].namespace
+tmpl_api_version := "crownlabs.polito.it/v1alpha2"
 
-ns = t_ns {
-	count([t_ns]) > 0
-	
-}
+tmpl_ref_field := "template.crownlabs.polito.it/TemplateRef"
 
-else = "default" {
-	true
+# Return the referenced template namespace, in case the field is defined
+get_tmpl_namespace(review) = namespace {
+	review.object.spec[tmpl_ref_field].namespace
+	namespace := review.object.spec[tmpl_ref_field].namespace
 }
 
-violation[{"msg": msg, "details": {}}] {
-	not data.inventory.namespace[ns]
-	msg := sprintf("Namespace %v doesn't exist", [data.inventory])
+# Return default, in case the namespace field is not defined
+get_tmpl_namespace(review) = namespace {
+	not review.object.spec[tmpl_ref_field].namespace
+	namespace := "default"
```

Comment:
To avoid replicating code, maybe it could be a good idea to use a if-else construct instead of using the same function twice. 
```suggestion
	get_tmpl_namespace(review) = namespace {
	review.object.spec[tmpl_ref_field].namespace
	namespace := input.review.object.spec[tmpl_ref_field].namespace 
    }else = "default"{
	true
   }
```

Reply:
Thanks, I modified the rule according to your suggestion. I honestly feel both solutions are pretty equivalent (also in terms of LOC), but it definitely makes sense to use the if-else construct.